src-java/reroute-topology/reroute-storm-topology/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteQueueService.java

```diff
@@ -228,15 +230,31 @@ public int compare(FlowThrottlingData throttlingDataA, FlowThrottlingData thrott
             int priorityB = throttlingDataB.getPriority() == null ? defaultFlowPriority : throttlingDataB.getPriority();
             Instant timeCreateA = throttlingDataA.getTimeCreate();
             Instant timeCreateB = throttlingDataB.getTimeCreate();
+            PathComputationStrategy pathComputationStrategyA = throttlingDataA.getPathComputationStrategy();
+            PathComputationStrategy pathComputationStrategyB = throttlingDataB.getPathComputationStrategy();
 
-            if (priorityA == priorityB && (timeCreateA != null || timeCreateB != null)) {
-                if (timeCreateA == null) {
-                    return -1;
+            if (priorityA == priorityB) {
+                if (pathComputationStrategyA == COST_AND_AVAILABLE_BANDWIDTH
+                        && pathComputationStrategyB == COST_AND_AVAILABLE_BANDWIDTH) {
+                    return Long.compare(throttlingDataB.getBandwidth(), throttlingDataA.getBandwidth());
```

Comment:
What if we have the same bandwidths? Shouldn't we then compare by creation time?

Reply:
Fixed